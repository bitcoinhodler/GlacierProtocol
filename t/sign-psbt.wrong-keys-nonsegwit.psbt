cHNidP8BAFMCAAAAAbJaEJtR1VI+2kmue66fWS44PgLQ0xvcbYtqmT6yWBX6AQAAAAD/////ASJxiAIAAAAAF6kUhUbnR8gJQ3awoK9EvFu7jdTwIY2HAAAAAAABAIoCAAAAAb9pizGVzRE0+oz591XjOsw9v8WnbJX3T+f6INm6U0pyAQAAABcWABQkpTNe3QyXFf5jXuKP/Dw1ruyJlv////8CEH6IAgAAAAAXqRSHANocFnGWreti1KXd+UCiep/cb4egf4gCAAAAABepFL05lruabEAdp+ZdTbs6m5eM0EXshwAAAAABBItSIQPRTdz7aBf1V5aVu7PrPhhYh78pQrAx5vcWNDuP5+no4iECj81G+GFLLL8xgJaWgkKh4ivPttjyttyTnIwnw0eyk3shAxWstVASD0zctGDVxJCAq1CMpMzY3t7Kwi/urIzQF9TBIQIrBj7i8i+eGYLBQP53hnLGgxEaeW3Mzb1Hu2Xj1gipg1SuAAA=
