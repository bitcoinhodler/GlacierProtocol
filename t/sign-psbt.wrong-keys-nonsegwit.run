#!/bin/bash

# Test validation of extra keys. 3rd key does not belong to this address.


$GLACIERSCRIPT --regtest=$1 sign-psbt << INPUT 2>&1
y
y
y
y
y
y
../../t/sign-psbt.wrong-keys-nonsegwit.psbt
y
4
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4
cSfvoUsWWZ81M4qGYciPHdNxHXQxPEsXBdGWjYdchpisPabtiKVQ
INPUT

retVal=$?
if [ $retVal -eq 0 ]; then
    echo "Error: expected glacierscript.py to fail"
    exit 1
fi
