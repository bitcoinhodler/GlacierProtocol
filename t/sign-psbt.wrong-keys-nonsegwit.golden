Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?Input a filename located in the current directory which contains the PSBT:
0.42500000 BTC in unspent inputs from cold storage address 2NAVkgszAvjZZwWpHpbCaxSaNqCCXPD8Bts
0.42496290 BTC going to destination address 2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow
Fee amount: 0.00003710
Confirm? (y/n): 
How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: Key #3: ERROR: that key does not belong to this source address. Exiting...
